src/sumBy.ts

```diff
@@ -1,4 +1,11 @@
 import { purry } from "./purry";
+import { type IterableContainer } from "./internal/types";
+
+type SumByBigInt<T, TS extends IterableContainer<T>> = TS extends []
+  ? 0
+  : TS extends readonly [T, ...ReadonlyArray<T>]
```

Comment:
We only care about the shape of the array, not it's content, the second generic is redundant here.

```suggestion
type SumByBigInt<T extends IterableContainer> = T extends []
  ? 0
  : T extends readonly [unknown, ...ReadonlyArray<unknown>]
```

Reply:
Done